includes/class-search-categories-metabox.php

```diff
@@ -0,0 +1,281 @@
+<?php
+
+class Search_Categories_Metabox {
+
+	/**
+	 * @const PLUGIN_SLUG
+	 */
+	CONST PLUGIN_SLUG = 'search-categories-metabox';
+
+	/**
+	 * @var string $plugin_file
+	 */
+	protected $plugin_file;
+
+	/**
+	 * @var string $plugin_version
+	 */
+	public $plugin_version = '0.1.0';
+
+	/**
+	 * Search_Categories_Metabox constructor.
+	 * @param $file
+	 */
+	public function __construct( $file ) {
+		$this->plugin_file = $file;
+	}
+
+	/**
+	 * Gets the Post taxonomy Category slug by Post ID.
+	 *
+	 * @since 0.1.0
+	 * @access private
+	 *
+	 * @param $post_id
+	 * @return array
+	 */
+	private function get_post_taxonomy( $post_id ) {
+		$taxonomies = get_post_taxonomies( $post_id );
```

Comment:
`get_post_taxonomies` can return more than one result and the order in which they are sorted is not guaranteed. Then how are you picking up just the first element in the array?